docs/maven-plugin/src/main/java/org/keycloak/guides/maven/GuideParser.java

```diff
@@ -0,0 +1,79 @@
+package org.keycloak.guides.maven;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class GuideParser {
+
+    private Pattern TMPL_IMPORT = Pattern.compile("<#import \"/templates/guide.adoc\" as (?<importName>[^ ]*)>");
+    private Pattern ELEMENT = Pattern.compile("(?<key>priority|title|summary)=(\\\"(?<valueString>[^\\\"]*)\\\"|(?<valueInt>[\\d]*))");
+
+    /**
+     * Parses a FreeMarker template to retrieve Guide attributes
+     * @param file
+     * @return A Guide instance; or <code>null</code> if not a guide
+     * @throws IOException
+     */
+    public Guide parse(File file) throws IOException {
+        try (BufferedReader br = new BufferedReader(new FileReader(file))) {
+            String importName = null;
+            String l = br.readLine();
+            // Read max 10 lines to find template import
+            for (int i = 0; i < 10 && importName == null; i++) {
```

Comment:
Could you elaborate a bit on why exactly 10?

Reply:
Figured that it would be in one of the first 10 lines, or not be there. So, just a random number trying to be supportive of it being not the "first" line, but then also limit how many lines are read looking for it.